che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitMergeResult.java

```diff
@@ -0,0 +1,112 @@
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - API 
+ *   SAP           - initial implementation
+ *******************************************************************************/
+package org.eclipse.che.git.impl.jgit;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.che.api.git.shared.MergeResult;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.merge.ResolveMerger;
+
+/**
+ * Jgit implementation of {@link MergeResult}
+ *
+ * @author Andrey Parfonov
+ */
+class JGitMergeResult implements MergeResult {
+    private final org.eclipse.jgit.api.MergeResult jgitMergeResult;
+
+    /**
+     * @param jgitMergeResult
+     *         back-end instance
+     */
+    JGitMergeResult(org.eclipse.jgit.api.MergeResult jgitMergeResult) {
+        this.jgitMergeResult = jgitMergeResult;
+    }
+
+    @Override
+    public String getNewHead() {
+        ObjectId newHead = jgitMergeResult.getNewHead();
+        if (newHead != null) {
+            return newHead.getName();
+        }
+        // Merge failed.
+        return null;
+    }
+
+    @Override
+    public MergeStatus getMergeStatus() {
+        switch (jgitMergeResult.getMergeStatus()) {
+            case ALREADY_UP_TO_DATE:
+                return MergeStatus.ALREADY_UP_TO_DATE;
+            case CONFLICTING:
+                return MergeStatus.CONFLICTING;
+            case FAILED:
+                return MergeStatus.FAILED;
+            case FAST_FORWARD:
+                return MergeStatus.FAST_FORWARD;
+            case MERGED:
+                return MergeStatus.MERGED;
+            case NOT_SUPPORTED:
+                return MergeStatus.NOT_SUPPORTED;
+            case CHECKOUT_CONFLICT:
+                return MergeStatus.CONFLICTING;
+        }
+        throw new IllegalStateException("Unknown merge status " + jgitMergeResult.getMergeStatus());
+    }
+
+    @Override
+    public List<String> getMergedCommits() {
+        ObjectId[] jgitMergedCommits = jgitMergeResult.getMergedCommits();
+        List<String> mergedCommits = new ArrayList<>();
+        if (jgitMergedCommits != null) {
+            for (ObjectId objectId : jgitMergedCommits) {
+                mergedCommits.add(objectId.getName());
+            }
+        }
+        return mergedCommits;
+    }
+
+    @Override
+    public List<String> getConflicts() {
+        if (jgitMergeResult.getMergeStatus().equals(org.eclipse.jgit.api.MergeResult.MergeStatus.CHECKOUT_CONFLICT)) {
+            return jgitMergeResult.getCheckoutConflicts();
+        }
+        Map<String, int[][]> conflicts = jgitMergeResult.getConflicts();
+        List<String> files = null;
```

Comment:
And specify it in java doc of [MergeResult](https://github.com/codenvy/che-core/blob/jgit/platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/shared/MergeResult.java#L50)